lib/ansible/cli/inventory.py

```diff
@@ -172,7 +172,7 @@ def dump(stuff):
         if context.CLIARGS['yaml']:
             import yaml
             from ansible.parsing.yaml.dumper import AnsibleDumper
-            results = yaml.dump(stuff, Dumper=AnsibleDumper, default_flow_style=False)
+            results = to_text(yaml.dump(stuff, Dumper=AnsibleDumper, default_flow_style=False, allow_unicode=True))
```

Comment:
use to_text on the return, then all cases will be covered

Reply:
Others already return `text`, though doing `to_text` on return might cover new formats, if we add them.

The remaining test case that fails is `--list --toml`. `toml` does some magic on Python 2 and escapes `\u1234` to `\\u1234` and then we display `\u1234` instead of the actual character. I am not sure I want to fight with this just to fix it for Python 2...

Reply:
Since it's Py2, and controller, we should just ignore py2.

Reply:
it is a consideration for the backport

Reply:
Since toml doesn't support any configuration for unicode, I wouldn't worry with it either way.